guides/howtos/Composable transactions with Multi.md

```diff
@@ -0,0 +1,210 @@
+      Repo.insert!(%Transfer{from: mary.id, to: john.id, amount: 10})
+    {:error, error} ->
+      Repo.rollback(error)
+  end
+end)
+```
+
+The snippet above starts a transaction and then calls `transfer_money/3` that also runs in a transaction. In case of multiple transactions, they are all flattened, which means a failure in an inner transaction causes the outer transaction to also fail. That's why matching and rolling back on `{:error, error}` is important.
+
+While nesting transactions can improve the code readability by breaking large transactions into multiple smaller transactions, there is still a lot of boilerplate involved in handling the success and failure scenarios. Furthermore, composition is quite limited, as all operations must still be performed inside transaction blocks.
+
+A more declarative approach when working with transactions would be to define all operations we want to perform in a transaction decoupled from the transaction execution. This way we would be able to compose transactions operations without worrying about its execution context or about each individual success/failure scenario. That's exactly what `Ecto.Multi` allows us to do.
+
+## Composing with data structures
+
+Let's rewrite the snippets above using `Ecto.Multi`. The first snippet that transfers money between mary and john can rewritten to:
+
+```elixir
+mary_update = from Account, where: [id: ^mary.id], update: [inc: [balance: +10]]
+john_update = from Account, where: [id: ^john.id], update: [inc: [balance: -10]]
+Ecto.Multi.new
+|> Ecto.Multi.update_all(:mary, mary_update)
+|> Ecto.Multi.update_all(:john, john_update)
+```
+
+`Ecto.Multi` is a data structure that defines multiple operations that must be performed together, without worrying about when they will be executed. `Ecto.Multi` mirrors most of the `Ecto.Repo` API, with the difference each operation must be explicitly named. In the example above, we have defined two update operations, named `:mary` and `:john`. As we will see later, the names are important when handling the transaction results.
+
+Since `Ecto.Multi` is just a data structure, we can pass it as argument to other functions, as well as return it. Assuming the multi above is moved into its own function, defined as `transfer_money(mary, john, value)`,  we can add a new operation to the multi that logs the transfer as follows:
+
+```elixir
+transfer_money(mary, john, 10)
+|> Ecto.Multi.insert(:transfer, %Transfer{from: mary.id, to: john.id, amount: 10})
+```
+
+This is considerably simpler than the nested transaction approach we have seen earlier. Once all operations are defined in the multi, we can finally call `Repo.transaction`, this time passing the multi:
+
+```elixir
+transfer_money(mary, john, 10)
+|> Ecto.Multi.insert(:transfer, %Transfer{from: mary.id, to: john.id, amount: 10})
+|> Repo.transaction()
+|> case do
+  {:ok, %{transfer: transfer}} ->
+    # Handle success case
+  {:error, name, value, rolled_back_changes} ->
```

Comment:
I think `rolled_back_changes` is a misleading name. One of Multi steps can contain a function which result is not rolled back along with DB transaction.

Reply:
```suggestion
  {:error, name, value, changes_so_far} ->
```